lib/src/v3/variable_tokenizer.dart

```diff
@@ -0,0 +1,292 @@
+    while (!_isAtEnd) {
+      // Invariant: At this point, we're not in a special token sequence (like
+      // a string literal). The methods always consume the special token and
+      // handle adding the lexeme to the transformed output.
+      final startIndex = _index;
+      final charCode = _consume();
+
+      if (charCode == _variableCodeUnit) {
+        _startVariable(startIndex);
+        continue nextToken;
+      } else {
+        switch (charCode) {
+          case $slash:
+            // `/*`, block comment
+            if (_consumeIfMatches($asterisk)) {
+              _blockComment();
+              continue nextToken;
+            }
+            break;
+          case $minus:
+            // `--`, line comment
+            if (_consumeIfMatches($minus)) {
+              _lineComment();
+              continue nextToken;
+            }
+            break;
+          case $doubleQuote:
+            // Double quote has already been consumed, but is part of the identifier
+            _rewrittenSql.writeCharCode($doubleQuote);
+            _continueEscapedIdentifier();
+            continue nextToken;
+          case $singleQuote:
+            // Write start of string that has already been consumed
+            _rewrittenSql.writeCharCode($singleQuote);
+            _continueStringLiteral(enableEscapes: false);
+            continue nextToken;
+          case $e:
+          case $E:
+            if (_consumeIfMatches($singleQuote)) {
+              // https://cloud.google.com/spanner/docs/reference/postgresql/lexical#string_constants_with_c-style_escapes
+              _rewrittenSql
+                ..writeCharCode(charCode) // e or E
+                ..writeCharCode($singleQuote);
+
+              _continueStringLiteral(enableEscapes: true);
+              continue nextToken;
+            }
+        }
+      }
+
+      // This char has no special meaning, add it to the SQL string.
+      _rewrittenSql.writeCharCode(charCode);
+    }
+  }
+
+  /// After reading the start of a block comment, this methods skips the until
+  /// we find its end.
+  void _blockComment() {
+    // Look for `*/` sequence to end the comment
+    while (!_isAtEnd) {
+      final char = _consume();
+      if (char == $asterisk) {
+        if (_consumeIfMatches($slash)) return;
+      }
+    }
+  }
+
+  /// After reading the start of a line comment, this method skips the current
+  /// source line.
+  void _lineComment() {
+    // Consume the rest of the line without adding it.
+    while (!_isAtEnd) {
+      final char = _consume();
+      if (char == $cr || char == $lf) return;
+    }
+  }
+
+  /// After reading the start of a string literal, this method reads the
+  /// remainder of that literal and adds it to the transformed output.
+  ///
+  /// Handles "double single-quotes" as an escape sequence. If C-style escapes
+  /// are enabled, `\'` is also not considered to end the string.
+  void _continueStringLiteral({required bool enableEscapes}) {
```

Comment:
I'm wondering how the postgres escape literals are handled, e.g. `U&"\0441\043B\043E\043D"` or  other backslash escape sequences. Could you please add a few more to the variable_tokenizer_test.dart to check if these are not in conflict with the parsing?

https://www.postgresql.org/docs/current/sql-syntax-lexical.html